### PR TITLE
feat: adding keydown event listeners for tabs

### PR DESCRIPTION
**Pull request by Ibabalola**
// 'Left', 'Right', 'Up' and 'Down' event listeners added for `<Tab />`

### CODE REVIEW

File: src/tabGroup/TabGroup.tsx [hunk 1]
```diff
@@ -65,6 +65,14 @@ type TabContextProps = {
    * Tab Context select tab
    */
   activeTab: string;
+  /**
+   * Tab previous tab
+   */
+  previousTab?: string;
```
Comment (alexwbbr) [inline src/tabGroup/TabGroup.tsx:71]: I'm not sure these need to be properties as you are calculating the previous and next tabs if these are blank anyway. Is there a scenario where having these as properties is beneficial?

**Review by alexwbbr — COMMENTED** [review]
Can you please also update the story to say that this feature has been added / how it works

**Comment by Ibabalola** [comment]
> Can you please also update the story to say that this feature has been added / how it works

Hi @alexwbbr , the change made here is internal, with the additional props added on the `type TabContextProps` which is for the `TabContext.Provider` therefore we have not exposed any new properties for the end consumer therefore I'm not how beneficial a story would be here.

The change made is simply to allow keyboard only users to be able to switch to the inactive tabs via keyboard controls

File: src/tabGroup/__test__/TabGroup.test.tsx [hunk 2]
```diff
@@ -330,6 +330,94 @@ describe('TabGroup', () => {
...
+
+    fireEvent.keyDown(tabs[0], { key: 'Right', code: 'Right', charCode: 39 });
+
+    expect(tabs[1]).toBeInTheDocument();
+    expect(tabItems[1]).toBeInTheDocument();
+    expect(tabItems[1].getAttribute('class')).toBe('tab-class-active');
+  });
+
+
+  it('should handle a keyDown right', () => {
```
Comment (alexwbbr) [inline src/tabGroup/__test__/TabGroup.test.tsx]: is this a duplicate test?

File: src/tabGroup/TabGroup.tsx (hunk 1 above)
Comment (Ibabalola) [reply src/tabGroup/TabGroup.tsx:71]: Indeed @alexwbbr , they are calculated in `<TabGroup />` but used in `<Tab>` to allow the active tab to change on 'Left', 'Right', 'Up' and 'Down' keydown event listeners.

I'm not entirely sure of an alternate approach and not aware of any adverse impact with the use of properties in this context.

**Comment by alexwbbr** [comment]
How would dev's know that the component has this functionality unless they read the change log then? I still think it should be mentioned somewhere that this has been added to the component

File: src/tabGroup/__test__/TabGroup.test.tsx (hunk 2 above)
Comment (Ibabalola) [reply src/tabGroup/__test__/TabGroup.test.tsx]: No, one of them is for tabbing to disable tab e.g. `<Tab eventKey="tab-2-id" label="tab 2 label" disabled>` however I have updated the description to clearly differentiate the tests.

**Comment by Ibabalola** [comment]
> How would dev's know that the component has this functionality unless they read the change log then? I still think it should be mentioned somewhere that this has been added to the component

added a keyboard event story @alexwbbr.

**Review by alexwbbr — APPROVED** [review]

**Pull request merged**